src/java.base/share/classes/java/nio/X-Buffer.java.template

```diff
@@ -258,6 +258,15 @@ import jdk.internal.util.ArraysSupport;
  * cb.put("text/").put(subtype).put("; charset=").put(enc);</pre></blockquote>
  *
 #end[char]
+ * <h2> Optional operations </h2>
+ * Methods specified as
+ * <i><a href="Buffer.html#read-only-buffers-heading">optional
+ * operations</a></i> throw a {@linkplain ReadOnlyBufferException} when invoked
+ * on a {@linkplain java.nio.Buffer#isReadOnly read-only} $Type$Buffer. The
+ * methods {@linkplain #array array} and {@linkplain #arrayOffset arrayOffset},
+ * however, instead throw an {@linkplain UnsupportedOperationException} if the
+ * $Type$Buffer is not backed by an {@linkplain Buffer#hasArray accessible
+ * $type$ array} irrespective of whether the $Type$Buffer is read-only.
```

Comment:
This mostly looks okay but I think it would be clearer to drop "however, instead"  and put "irrespective ..." in parentheses.

Reply:
I thought "however, instead" a bit clunky as well; will update.

Reply:
Refined in 12e5fe15237a5f5cebbaf2ed4e81a851074e3e5c.